include/rocksdb/public_port.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+#pragma once
+
+// Best used after all the headers in case windows #defines give you trouble
+// Minimum subset
+
+#ifndef ROCKSDB_PUBLIC_PORT_H__
+#define ROCKSDB_PUBLIC_PORT_H__
+
+#ifdef OS_WIN
```

Comment:
This will not work in public header files since public header files are not compiled. We don't pass the defines to our customers.